src/components/pages/dashboard/dashboard.js

```diff
@@ -124,7 +124,7 @@ class DashboardPage extends Component {
           <Grid fluid className="layout">
             <Row>
               <Col md={8}>
-                <DeviceMap {...deviceMapProps} />
+                {this.props.BingMapKey ? <DeviceMap {...deviceMapProps}/> : null} 
```

Comment:
Doesn't it make more sense to place the checks for the key in the device map component itself? Is that difficult for some reason?

Reply:
Complete deviceMap is based on the BingMapkey whether it is static or actual key. Previously it used to be there but i removed last twodays back.